src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -196,18 +196,21 @@ public static Cap parseCap(String cap) throws ParseException {
             requireNonNull(cap);
             String trimmedCap = cap.trim();
             String[] capValues = trimmedCap.split(CAP_SEPARATOR);
-            double capValue;
-            double maximumCapValue;
-            capValue = Double.parseDouble(capValues[0]);
-            maximumCapValue = Double.parseDouble(capValues[1]);
+            if (!Cap.isValidCapFormat(capValues[0]) || !Cap.isValidCapFormat(capValues[1])) {
+                throw new ParseException(Cap.MESSAGE_NUMERIC_VALUE_REQUIRED);
+            } else if (capValues.length > 2) { // more than two values provided
+                throw new ParseException(Cap.MESSAGE_INVALID_CAP_FORMAT);
+            }
```

Comment:
Minor nitpick, feel u can refactor this whole portion out into a single method that throws a ParseException to improve code quality further